### ISSUE TITLE
Add capability to permit only configured IP addresses to access a site/tenant

### ISSUE THREAD

**Issue by CrispinF** [cloudscribe/cloudscribe#1011]
Counterpoint to #918, but for specifying permitted IP addresses that _can_ access a site/tenant.

Obviously risks locking everyone out of a site when configured, so perhaps ensure this does not apply on /siteadmin/ and child pages...? No, that undermines the whole point of this, which is to improve security. If an admin locked themselves out of a site accidentally they could fix this in the database.

UI could usefully show the user's current IP, to assist in configuring.

Once a permitted IP is configured, requests from any other IP should get a 403 Forbidden response (discuss? it's configurable in IIS).

(I don't think we need to consider IP ranges/masks at this stage).

**Comment by CrispinF** [comment]
@markidox @SimonAnnetts all Simon's suggestion sound good, but please don't hardcode any help links to online resources. We can put such things in the cloudscribe documentation pages which are easily updated.
Not sure about 8 - is there a reliable way of doing this? I think we have to presume any users of this tool are experts, and do not need to mollycoddle, especially if that itself is complex and/or unreliable.

**Comment by markidox** [comment]
@SimonAnnetts @CrispinF No worries. I've done 1,2,3. I'm unsure about point 4, the files are already pretty large, combining them all is going to make it, obviously, even larger. It also makes it pretty easy to follow that they're separate...

For 6, Why would you want to see the resolved range? Surely you know what the range is otherwise why are you entering it?

I'll send you a link to the draft docs @SimonAnnetts so you can see what we've produced so far in that regard.

**Comment by JimKerslake** [comment]
> For 6, Why would you want to see the resolved range? Surely you know what the range is otherwise why are you entering it?

I'd be looking for some help on the UI about your supported format(s) for entering an IP range, because I've never really fount it at all obvious that 192.168.1.18/28 means the same as 192.168.1.18/255.255.255.240 means the same as 192.168.1.16-192.168.1.31
(had to use AI to come up with that example)

So maybe something on the UI to tell me which of above 3 format(s) are supported?

**Comment by markidox** [comment]
Fair enough, there are examples in the documentation, I guess there's no harm in surfacing those on the FE:

The following range formats are accepted:

192.168.0.0/255.255.255.0
192.168.0.10 - 192.168.10.20
192.168.10.10-20
fe80::/10
192.168.0.0/16

**Comment by SimonAnnetts** [comment]
re point 6, I just think it's a useful belt and braces check to be able to see and validate how the code is going to resolve the address/CIDR that you have input. You could send the input of the IP address box to a simple API endpoint that just returns the resolved range (first and last IP) and displays it in a label underneath the input box when you press the button.

**Comment by markidox** [comment]
OK. I've done 1,2,3,4 and (part of) 9. For points 5, (part of) 6 and (the other part of) 9 - these will be covered off in the documentation on the website and I've copied the IP Range examples onto the front end too.

For the rest of 6, that actual resolver, I'm still unsure but perhaps something to look at for phase 2? 7 I think needs a discussion.

**Comment by markidox** [comment]
Yeah, if the nav doesn't go up as part of the merge it'll need doing.
